src/golang/cmd/server/handler/configure_storage.go

```diff
@@ -0,0 +1,140 @@
+	OperatorRepo       repos.Operator
+
+	PauseServer   func()
+	RestartServer func()
+}
+
+type configureStorageArgs struct {
+	*aq_context.AqContext
+	// This is the ID of the integration to use as the new storage layer.
+	// It should only be set if configureLocalStorage is false.
+	storageIntegrationID  uuid.UUID
+	configureLocalStorage bool
+}
+
+func (*ConfigureStorageHandler) Name() string {
+	return "ConfigureStorage"
+}
+
+func (h *ConfigureStorageHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, errors.Wrap(err, "Unable to configure storage layer.")
+	}
+
+	integrationIDStr := chi.URLParam(r, routes.IntegrationIdUrlParam)
+
+	if integrationIDStr != "local" {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "We currently only support changing the storage layer to the local filesystem from this route.")
+	}
+
+	return &configureStorageArgs{
+		AqContext: aqContext,
+		// TODO ENG-2574: Add support for switching to non-local storage
+		storageIntegrationID:  uuid.Nil,
+		configureLocalStorage: true,
+	}, http.StatusOK, nil
+}
+
+func (h *ConfigureStorageHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*configureStorageArgs)
+
+	// TODO ENG-2574: Remove this assumption
+	if !args.configureLocalStorage {
+		return nil, http.StatusBadRequest, errors.New("We currently only support changing the storage layer to the local filesystem from this route.")
+	}
+
+	currentStorageConfig := config.Storage()
+
+	if currentStorageConfig.Type == shared.FileStorageType {
+		return nil, http.StatusBadRequest, errors.New("The storage layer is already set to the local filesystem.")
+	}
+
+	newStorageConfig := shared.StorageConfig{
+		Type: shared.FileStorageType,
+		FileConfig: &shared.FileConfig{
+			Directory: path.Join(config.AqueductPath(), "storage"),
+		},
+	}
+
+	log.Info("Starting storage migration process...")
+
+	// Wait until the server is paused
+	h.PauseServer()
+	// Makes sure that the server is restarted
+	defer h.RestartServer()
+
+	// Wait until there are no more workflow runs in progress
+	lock := utils.NewExecutionLock()
+	if err := lock.Lock(); err != nil {
+		log.Errorf("Unexpected error when acquiring workflow execution lock: %v. Aborting storage migration!", err)
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to migrate to the new storage layer")
+	}
+	defer func() {
+		if err := lock.Unlock(); err != nil {
+			log.Errorf("Unexpected error when unlocking workflow execution lock: %v", err)
+		}
+	}()
+
+	// Migrate all storage content to the new storage config
+	if err := utils.MigrateStorageAndVault(
+		ctx,
+		&currentStorageConfig,
+		&newStorageConfig,
+		args.OrgID,
+		h.DAGRepo,
+		h.ArtifactRepo,
+		h.ArtifactResultRepo,
+		h.OperatorRepo,
+		h.IntegrationRepo,
+		h.Database,
```

Comment:
do we need a transaction here?

Reply:
Nope, the function being called performs operations that need to be atomic inside of a transation.